### ISSUE TITLE
Could not download EdgeDriver of given version from Microsoft repository

### ISSUE THREAD

**Issue by mitul494** [bonigarcia/webdrivermanager#444]
**Description of the problem**: 
It could not downloaded given version of Microsoft Edge(Chromium) from Microsoft repository
https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/
                
I wrote only simple line as following:
edgedriver().version("79.0.309.54").setup();

More on this, this version is found in "Full Version Directory" link at Microsoft repository page of Edge [URL# https://msedgewebdriverstorage.z22.web.core.windows.net/] and also tried with following as setEdgeDriverMirrorURL method is not available, but could not work;

edgedriver().config().setEdgeDriverUrl(new URL("https://msedgewebdriverstorage.z22.web.core.windows.net/"));

or

edgedriver().config().setEdgeDriverUrl(new URL("https://msedgewebdriverstorage.z22.web.core.windows.net/?prefix=79.0.309.54"));  

//---
edgedriver().fordeDownload().version("79.0.309.54").setup();

**Browser and version**: 
Edge (Chromium) Version 80.0.361.62 (Official build) (64-bit)

**Operating system**: 
Windows 10, 64 bit

**WebDriverManager version**:
3.8.1

**WebDriverManager traces**: 
2020-02-28 17:00:47 INFO  frameworkLogger:32 - BaseTest --- Setup called for browser [Edge]
[main] ERROR io.github.bonigarcia.wdm.WebDriverManager - msedgedriver 79.0.309.54 for WIN64 not found in https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/
[main] WARN io.github.bonigarcia.wdm.WebDriverManager - There was an error managing msedgedriver 79.0.309.54 (msedgedriver 79.0.309.54 for WIN64 not found in https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/) ... trying again using mirror
[main] ERROR io.github.bonigarcia.wdm.WebDriverManager - msedgedriver 79.0.309.54 for WIN64 not found in https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/
[main] WARN io.github.bonigarcia.wdm.WebDriverManager - There was an error managing msedgedriver 79.0.309.54 (msedgedriver 79.0.309.54 for WIN64 not found in https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/) ... trying again using latest from cache
[main] INFO io.github.bonigarcia.wdm.WebDriverManager - Latest version of Edge driver is 82.0.438.0
[main] INFO io.github.bonigarcia.wdm.Downloader - Downloading https://msedgedriver.azureedge.net/82.0.438.0/edgedriver_win64.zip

**Comment by bonigarcia** [comment]
This line is not going to work:

```
edgedriver().config().setEdgeDriverUrl(new URL("https://msedgewebdriverstorage.z22.web.core.windows.net/"));
```

... because as of now, WebDriverManager read the version parsing this [page](https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/). I plan to change this behavior in future versions of WebDriverManager. I'll let you know when it is available.

**Comment by eliasnogueira** [comment]
We could use only the https://msedgewebdriverstorage.z22.web.core.windows.net URL but the problem is we cannot determine the latest properly. The `LAST_STABLE` file has a not so old Edge version, but it's not the latest

Example: if you hit the following URL, you'll see
https://msedgedriver.azureedge.net/LATEST_STABLE
```
80.0.361.69
```
My guess is that happened because version 80 has the latest push on that storage...
If you compare the _last modified_ timestamp on the end of the storage list the `LATEST_RELEASE_80` has the same timestamp as `LATEST_STABLE`

I think we could use one of the following approaches:

**Approach 1**
If my guess is right it was, probably, a mistake on the storage and the next new versions will be added, automatically fixing the `LAST_STABLE` file. But we don't have any guarantee and, implementing the approach to use only the storage URL might show problems to the user if he wants to use the latest.

**Approach 2**
Maybe we might add a mix of both: use the current implementation, that gets the latest from https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver and, after, switches to the storage URL adding the version set explicitly by the user.

Let me know if you need some help @bonigarcia

**Comment by eliasnogueira** [comment]
Code snippet for the storage approach

```
private String getLatestVersion() throws MalformedURLException {
    String latestVersion = null;
    URL latestVersionUrl = new URL(MessageFormat.format("{0}/{1}", BASE_URL, LATEST_REFERENCE));

    try (BufferedReader reader = new BufferedReader(new InputStreamReader(latestVersionUrl.openStream(),
        StandardCharsets.UTF_16))) {
        latestVersion = reader.readLine().trim();
    } catch (IOException e) {
        // logging strategy here
        e.printStackTrace();
    }
    return latestVersion;
}

/**
 * @param version      browser version like 82.0.458.0
 * @param os           win | mac
 * @param architecture 32 | 64
 */
private void downloadSpecificVersion(String version, String os, String architecture) {
    /*
     * String version = "82.0.458.0";
     * String os = "mac";
     * String architecture = "64";
     *
     * downloadSpecificVersion(version, os, architecture);
     */

    String urlToDownloadFile = MessageFormat
        .format("{0}/{1}/{2}{3}{4}{5}", BASE_URL, version, PREFIX_FILENAME, os, architecture, ZIP);
    System.out.println("urlToDownloadFile = " + urlToDownloadFile);
}
```

**Comment by bonigarcia** [comment]
I think the `LATEST_STABLE` information is actually updated, isn't it? I have just checked in an Edge browser, in Windows:

![image](https://user-images.githubusercontent.com/5413013/77316828-63d18f00-6d0a-11ea-9bbc-050ec3ce1d05.png)

@eliasnogueira Why do you say it is not the latest version?

Regarding the proposed approaches, my idea was to use the `LATEST_STABLE` info to find out the latest version of the Chromium-based Edge (i.e. version 80+) and continue parsing the content of https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/ for the legacy versions (1.10240 to 6.17134).

@eliasnogueira Please make a PR about this if you want. Thanks a lot!

**Comment by eliasnogueira** [comment]
Actually you're right. My bad.
I'll create a PR implementing this approach.

**Comment by eliasnogueira** [comment]
@mitul494 Could you do the same test on your side?
A change was made and now It's working.
Thanks!

**Comment by bonigarcia** [comment]
@eliasnogueira I made a big refactoring for the next major release of WebDriverManager. I plan to publish this release very soon. Could you please execute the complete test suite of the master branch in your Mac machine to check everything is ok? Thank you in advance.

**Comment by eliasnogueira** [comment]
Sorry @bonigarcia , I didn't have a chance to comment about it before because my PR was gone  :)
I'll open a ticket and add the results there to track all the information about this regression test. What do you think?

**Comment by bonigarcia** [comment]
This feature is available in WebDriverManager 4.0.0, recently released.
